File: Programming Basics/Python/SortingAlgorithms/quicksort.py
def quick_sort(arr):
    length = len(arr)
    if length <= 1:
        return arr
    else:
        pivot = arr.pop()
        left_half = []
        right_half = []
        for num in arr:
            if num > pivot:
                right_half.append(num)
            else:
                left_half.append(num)

    return quick_sort(left_half) + [pivot] + quick_sort(right_half)


print(quick_sort([3, 4, 5, 6, 3, 2, 4, 1, 9]))

File: Programming-Basics/C++/sort/quicksort.cpp
#include<iostream>
 using namespace std;

 void swap(int arr[],int i,int j)
 {
 	int temp=arr[i];
 	arr[i]=arr[j];
 	arr[j]=temp;
 }

 int partition(int arr[],int l,int r)
 {
 	int pivot=arr[r];
 	int i=l-1;
 	int j;
 	for( j=l;j<r;j++)
 	{
       if(arr[j]<pivot)
 	  {
 		  i++;
 		  swap(arr,i,j);
 	  }
 	}
 	swap(arr,i+1,r);
 	return i+1; //returning the new pivot location after partition
 }

 void quicksort(int arr[],int l,int r)
 {
 	if(l<r)
 	{
     int pi=partition(arr,l,r);
 	   quicksort(arr,l,pi-1);
 	   quicksort(arr,pi+1,r);
 	}

 }



 int main()
 {
 	int n;
 	cin>>n;
 	int arr[n];
 	for(int i=0;i<n;i++)
 	{
 		cin>>arr[i];
 	}
 	quicksort(arr,0,n-1);
 	for(int i=0;i<n;i++)
 	{
 		cout<<arr[i]<<" ";
 	}
 	cout<<endl;
 	return 0;
 }
